packages/devtools_app/lib/src/screens/inspector_v2/inspector_controller.dart

```diff
@@ -731,37 +732,26 @@ class InspectorController extends DisposableController
     final pendingSelectionFuture = group.getSelection(
       selectedDiagnostic,
       treeType,
+      // If implementation widgets are hidden, the only widgets in the tree are
+      // those that were created by the local project.
+      inLocalProject: implementationWidgetsHidden.value,
```

Comment:
if `implementationWidgetsHidden.value` is false, then the selected widget may or may not be in the local project right? Can we still select a "local project node" if `inLocalProject` is false?

Reply:
Regardless of whether `implementationWidgetsHidden.value` is `true` (only local project widgets in tree) or `false` (local project and implementation widgets in tree), the node that the user selected may or may not be in the local project. This is because we allow a user to select any widget with the on-device widget selector. Maybe `inLocalProject` is a bad name (I can change it to `inLocalProjectOnly` or `restrictToLocalProject`?). Because yes we can select a local project node if `inLocalProject` is false.

How on-device selection works (I had to poke around a bit because this was unintuitive to me - still want to write up a one-pager at some point):

- a user selects a widget on their device
- the Flutter Widget Inspector sends an "inspect" event
- DevTools intercepts this "inspect" event and sends a `getSelection` request
- DevTools uses the response from `getSelection` to determine which node to select in the widget tree

So what this change is doing is modifying the request to `getSelection` to ask for the selected **local** widget (if implementation widgets are hidden in the tree) because we are unable to show the implementation widget in the tree.